### ISSUE TITLE
InvalidJavadocPositionCheck: false negative on javadocs with local variables 

### ISSUE THREAD

**Issue by rnveach** [checkstyle/checkstyle#6857]
Identified at https://github.com/checkstyle/checkstyle/pull/6718#discussion_r294065700 ,

Javadocs attached to a local variable are seen as valid because it thinks they are fields and it doesn't check if they are located in the class object token or not.
See https://github.com/checkstyle/checkstyle/blob/6bab0788365ba9afb713e14d9f6263f1f60b7aa2/src/it/resources/com/google/checkstyle/test/chapter7javadoc/rule711generalform/InputInvalidJavadocPosition.java#L44

`invalid` was used because IT area wouldn't allow `warn` since it doesn't produce a violation.

Extra example:
````
$ cat TestClass.java
public class TestClass {
    void method() {
/** Test */ int field;
    }
}

$ cat TestConfig.xml
<?xml version="1.0"?>
<!DOCTYPE module PUBLIC
          "-//Puppy Crawl//DTD Check Configuration 1.3//EN"
          "https://checkstyle.org/dtds/configuration_1_3.dtd">

<module name="Checker">
    <property name="charset" value="UTF-8"/>

    <module name="TreeWalker">
<module name="SummaryJavadoc" />
    </module>
</module>

$ java -jar checkstyle-8.22-all.jar -c TestConfig.xml TestClass.java
Starting audit...
[ERROR] TestClass.java:3: First sentence of Javadoc is missing an ending period. [SummaryJavadoc]
Audit done.
Checkstyle ends with 1 errors.
````

Expect no violation as local variable can't have javadocs.

**Comment by romani** [comment]
it is better to keep bug reports are defined in issue template, "all" jar can be build on snaphot version.

**Comment by rnveach** [comment]
Updated. At the time I didn't think there would be other violations in released checks. InvalidJavadocPosition isn't released yet so there was no way to do it with that check.

**Comment by rnveach** [comment]
Fix was merged
